### PR TITLE
[PAY-575] [PAY-574] [PAY-573] [PAY-572] [PAY-581] Coinbase Pay Button and Rewards Page Fixes

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

See commit messages.

https://user-images.githubusercontent.com/3690498/186288430-5c45f6a7-e8a1-4916-a2f2-b8e2f9d5a623.mov

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-design-fixes

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-design-fixes

File: packages/web/src/components/toggle-collapse-button/ToggleCollapseButton.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  className?: string
+  toggleButtonClassName?: string
+  showByDefault?: boolean
+  collapsedHeight?: number
+  showText: string
+  hideText: string
+  children: React.ReactNode
+}) => {
+  const [isCollapsed, setIsCollapsed] = useState(!showByDefault)
+  const handleToggle = useCallback(() => {
```
Comment (dylanjeffers) [inline packages/web/src/components/toggle-collapse-button/ToggleCollapseButton.tsx:30]: spaces if ya don't mind

File: packages/web/src/components/toggle-collapse-button/ToggleCollapseButton.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+export const ToggleCollapseButton = ({
+  className,
+  toggleButtonClassName,
+  showByDefault = false,
+  collapsedHeight = 0,
+  showText,
+  hideText,
+  children
+}: {
+  className?: string
```
Comment (dylanjeffers) [inline packages/web/src/components/toggle-collapse-button/ToggleCollapseButton.tsx:21]: also i personally like always breaking up the prop types into separate type, so `ToggleCollapseButtonProps`, just makes it easier to read and export when needed.

File: packages/web/src/components/toggle-collapse-button/ToggleCollapseButton.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    <div className={cn(className, { collapsed: isCollapsed })}>
+      <div
+        className={styles.toggleCollapsedContentsContainer}
+        style={{ height: isCollapsed ? collapsedHeight : bounds.height }}
+      >
+        <div ref={ref} className={styles.toggleCollapsedContents}>
+          {children}
+        </div>
+      </div>
+      <div
```
Comment (dylanjeffers) [inline packages/web/src/components/toggle-collapse-button/ToggleCollapseButton.tsx:47]: should ideally be a button, with an "aria-expanded" value, and an "aria-controls" to indicate which piece of content it is closing!

**Review by dylanjeffers — APPROVED** [review]
nice refactor

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-design-fixes

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-design-fixes

**Pull request merged**